homeassistant/components/tradfri/fan.py

```diff
@@ -43,31 +46,42 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Load Tradfri switches based on a config entry."""
-    gateway_id = config_entry.data[CONF_GATEWAY_ID]
-    tradfri_data = hass.data[DOMAIN][config_entry.entry_id]
-    api = tradfri_data[KEY_API]
-    devices = tradfri_data[DEVICES]
-
-    async_add_entities(
-        TradfriAirPurifierFan(dev, api, gateway_id)
-        for dev in devices
-        if dev.has_air_purifier_control
-    )
+    if (
+        COORDINATOR in hass.data[DOMAIN][config_entry.entry_id]
```

Comment:
The coordinator will always be here since the platform won't be setup until the integration entry is setup.